### PR TITLE
feat(angular): add scam generator for pipes and directives

### LINKED ISSUE #8122: add scam generator for pipes and directives
## Description

SCAMs can also be applied to pipes and directives with their own ngModule.

## Motivation

Treeshaking.

## Suggested Implementation

Working on PR

### PR DESCRIPTION
**Pull request by wizardnet972**
ISSUES CLOSED: #8122

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/angular/src/generators/scam-pipe/schema.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface Schema {
+  name: string;
+  path?: string;
+  project?: string;
+  skipTests?: boolean;
+  inlineScam?: boolean;
+  flat?: boolean;
+  prefix?: string;
```
Comment (Coly010) [inline packages/angular/src/generators/scam-pipe/schema.d.ts:8]: this is not a known argument for Pipes:
https://github.com/angular/angular\-cli/blob/master/packages/schematics/angular/pipe/schema.json

File: packages/angular/src/generators/scam-pipe/schema.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      "type": "boolean",
+      "description": "Create the NgModule in the same file as the Pipe.",
+      "default": true
+    },
+    "flat": {
+      "type": "boolean",
+      "description": "Create the new files at the top level of the current project.",
+      "default": true
+    },
+    "prefix": {
```
Comment (Coly010) [inline packages/angular/src/generators/scam-pipe/schema.json:47]: this is an incorrect argument for the Pipe schema. Remove it to match:
https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/pipe/schema.json

File: packages/angular/generators.json [hunk 3]
```diff
@@ -115,6 +115,16 @@
       "schema": "./src/generators/scam/schema.json",
       "description": "Generate a component with an accompanying Single Component Angular Module (SCAM)."
     },
+    "scam-directive": {
```
Comment (Coly010) [inline packages/angular/generators.json:118]: Both of these need to be added to the generators block of this JSON, pointing to the non-compat file.

Take a look at the `scam` entry further in the file to get an idea of how it's done.

**Review by Coly010 — CHANGES_REQUESTED** [review]
This looks good overall! Thanks for the PR!

There are some minor things that need adjusted :)

**Comment by wizardnet972** [comment]
@Coly010 Sorry something went wrong.. I'll send a new PR

**Pull request closed without merging**